### ISSUE TITLE
no error when dataclass has an `__init__` when `init=True`

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#343]
```py
from dataclasses import dataclass

@dataclass
class Bar:
    def __init__(self): # should be an error telling the user to put init=False in the dataclass decorator
        ...
```

it doesn't cause a runtime error but this implicit behavior is likely a mistake, especially if the user does specifies `init=True` explicitly:
```py
from dataclasses import dataclass

@dataclass(init=True)
class Bar:
    def __init__(self): # no error
        ...
```
